pkg/substitution/substitution.go

```diff
@@ -33,12 +45,12 @@ func FillConfigValues(config config.OperatorConfig, templates []client.Object) [
 		// Set namespaces for all object. Namespace on cluster-wide objects is stripped by API server and is not applied
 		templateCopy.SetNamespace(config.ManagedNamespace)
 
-		dep, ok := templateCopy.(*v1.Deployment)
-		if ok {
+		switch dep := templateCopy.(type) {
```

Comment:
Please pick another name, as we are already working with Deployments and DaemonSets.